### DISCUSSION TITLE
join private channel with invite link

### DISCUSSION THREAD

**Discussion by Phandorin** [gotd/td#1063] [Q&A] [answered]
Hi !
I can't join private channel with invite link. Trying "MessagesCheckChatInvite()" and "MessagesImportChatInvite()" functions but all time receive: "rpcDoRequest: rpc error code 400: INVITE_HASH_EXPIRED". Invite link absolutely alive and work in official messenger. 
Thanks for help!

**Comment by ernado** [comment]
https://github.com/gotd/td/blob/adfab92211b96e4f389d559888007bc962c70791/telegram/internal/deeplink/deeplink.go#L14-L39

**Comment by ernado** [comment, marked as answer]
[gotd/td/telegram/message#Sender.JoinLink](https://pkg.go.dev/github.com/gotd/td@v0.79.0/telegram/message#Sender.JoinLink)

**Reply by Phandorin** [reply]
Thank's, of course i use invite link in all accepted format, but take the same result. My error so strange: code - 400, but error describe " INVITE_HASH_EXPIRED" must be with code 406 in documentation.

**Comment by Phandorin** [comment]
Thanks!!! I understood my mistake) everything work!
